tests/http/test_proxyagent.py

```diff
@@ -100,22 +102,25 @@ def _make_connection(
 
         return http_protocol
 
-    def test_http_request(self):
-        agent = ProxyAgent(self.reactor)
+    def _test_request_no_proxy(self, agent, scheme, hostname, path):
```

Comment:
Why is this helper method called `*no_proxy` when we determine whether to use a proxy in the calling function?

Also could you add a docstring with a summary of the method's purpose please?

Reply:
Thanks, I've renamed this `_test_request_direct_connection` to try and disambiguate between `no_proxy` referring to the environment variable. I've also added a docstring to describe the method.